repository/src/org/pentaho/platform/repository2/unified/jcr/jackrabbit/security/SpringSecurityPrincipalProvider.java

```diff
@@ -115,9 +118,35 @@
    */
   private final AtomicBoolean initialized = new AtomicBoolean( false );
 
-  private final LRUMap userCache = new LRUMap( 4096 );
+  private LRUMap getRolesCache() {
+    return getCache( ROLE_CACHE, 512 );
+  }
+
+  private LRUMap getUserCache() {
+    return getCache( USER_CACHE, 4096 );
+  }
+
+  void setCacheManager( ICacheManager cacheManager ) {
+    this.cacheManager = cacheManager;
+  }
 
-  private final LRUMap roleCache = new LRUMap( 512 );
+  private LRUMap getCache( String region, int capacity ) {
```

Comment:
This is not how you use/create regions. In your code the region is simply a "key" into the global cache.

Also instead of simply storing the LRUMap in the cache. create a region and then use that region as the map: putInRegionCache(REGION, Username, User), etc.